packages/core-utils/src/time.js

```diff
@@ -48,12 +50,9 @@ export function formatDuration(seconds) {
  * @returns {string} formatted text representation
  */
 export function formatDurationWithSeconds(seconds) {
-  const dur = moment.duration(seconds, "seconds");
-  let text = "";
-  if (dur.hours() > 0) text += `${dur.hours()} hr, `;
-  if (dur.minutes() > 0) text += `${dur.minutes()} min, `;
-  text += `${dur.seconds()} sec`;
-  return text;
+  return formatDistanceStrict(0, 1000 * seconds, {
+    includeSeconds: true
+  });
```

Comment:
`includeSeconds` isn't a valid argument for `date-fns#formatDistanceStrict` and there is rounding that occurs that still makes the time fuzzy. Also, these methods don't output things quite right. Consider what happens with the input of `5400` (90 minutes) with the `formatDistanceStrict` and also `formatDuration` methods of `date-fns`:

```js
console.log(
  formatDistanceStrict(
    0, 
    seconds * 1000, 
    {
      unit: 'second'
    }
  )
) // ===> "5400 seconds"
console.log(
  formatDuration(
    {
      seconds
    },
    { format: ['hours', 'minutes', 'seconds'] }
  )
) // ===> "5400 seconds"
```

It does seem like this works though:

```js
const hours = Math.floor(seconds / 3600)
const minutes = Math.floor((seconds - hours * 3600) / 60)
const seconds2 = seconds - hours * 3600 - minutes * 60
console.log(
  formatDuration(
    {
      hours,
      minutes,
      seconds: seconds2
    },
    { 
      format: ['hours', 'minutes', 'seconds']
    }
  )
)
```

And, it is possible to add a custom locale object/function like so to get to the abbreviated names:

```js
const hours = Math.floor(seconds / 3600)
const minutes = Math.floor((seconds - hours * 3600) / 60)
const seconds2 = seconds - hours * 3600 - minutes * 60
const specLookup = {
  xHours: 'hr',
  xMinutes: 'min',
  xSeconds: 'sec'
}
console.log(
  formatDuration(
    {
      hours,
      minutes,
      seconds: seconds2
    },
    { 
      format: ['hours', 'minutes', 'seconds'],
      locale: { 
        code: 'en-US',
        formatDistance: (spec, val) => {
          return `${val} ${specLookup[spec]}`
        }
      }
    }
  )
)
```

However, adding in the locale like this may not even be needed since it seems like the direction in otp-react-redux is towards using a custom message component. See https://github.com/opentripplanner/otp-react-redux/blob/518def8f187b888d40dd7dcd9aced7e1af01f74f/lib/components/narrative/default/default-itinerary.js#L48-L70.

So with all that said, maybe the best option here is to use the final code snippet I posted.

Reply:
Your knowledge of how things are supposed to be is helpful here thank you! Have followed your suggestion and implemented your fix in 3bcbd9b4ee5744d97fb83f6041f757dc0d8007ae